### ISSUE TITLE
Carry over settings from regular deployment

### ISSUE THREAD

**Issue by koscejev** [paypal/resteasy-spring-boot#9]
According to javadoc, `CustomDispatcherResteasyDeployment` will likely be removed in the future, so this will not be an issue. But I don't know how likely it is that this will be incorporated into Resteasy itself anytime soon. :/

```
	public CustomDispatcherResteasyDeployment(ResteasyDeployment regularDeployment) {
		this.actualProviderClasses = regularDeployment.getActualProviderClasses();
	}
```

When `CustomDispatcherResteasyDeployment` is created, it's created based on regular `ResteasyDeployment`. However, the custom one only takes `actualProviderClasses` from the regular one. This means all other settings are ignored. For example, I can customize `server.context-parameters.resteasy.role.based.security=true` in application.properties and it'll properly set `securityEnabled` to `true` in regular deployment. But then this setting is not carried over to the custom one.

**Comment by koscejev** [comment]
Judging by #7 this really is non-issue - just have to wait for Resteasy 3.0.17.Final. So I'm guessing this will be closed as Won't Fix. 

Is there any workaround you could suggest in the mean time, so we can continue developing our wildfly/spring-boot migration PoC?

**Comment by fabiocarvalho777** [comment]
You are right, this is not considered an issue because CustomDispatcherResteasyDeployment (and the other two custom classes) are supposed to be removed as soon as RESTEasy 3.0.17.Final is released. By the way, notice that they have already merged [my PR](https://github.com/resteasy/Resteasy/pull/744), so there is no doubts RESTEasy 3.0.17.Final will have the necessary support so that we can remove the three custom classes.

We can follow up RESTEasy 3.0.17.Final timeline [in this page here](https://issues.jboss.org/projects/RESTEASY/versions/12329806). Notice that unfortunately they haven't set a target date yet.

So we have two options. We can either just wait until RESTEasy 3.0.17.Final is released, then I can release RESTEasy Spring Boot starter 2.0.0, or, we can go ahead and modify CustomDispatcherResteasyDeployment now, so that all ResteasyDeployment settings are carried over, then I can deliver a 2.0.0-SNAPSHOT version with this change.

I would say the decision depends on your timeline and how "blocking" this issue is for you right now.
Just let me know.

Thanks again for your contributions.

**Comment by koscejev** [comment]
The timeline for 3.0.17 is not encouraging - it seems they are only in the middle of this release and they do only about 4-5 releases a year, so this could easily take a month or two. However, I also don't like the idea of developing something that will definitely be thrown away completely, especially considering how many things one would have to think about to transfer all those settings.

Perhaps there's a 3rd option? I noticed that other spring-boot-starters have separate configuration, so they can pick up their custom configuration from spring-boot's `application.properties`. Is there a possibility to add custom configs for resteasy to `application.properties` to avoid having to use servlet context for configuration? I would guess that implies creating custom subclass of `ConfigurationBootstrap` similar to `ListenerBootstrap`, but one that wouldn't require `ServletContext` and would pick configuration from spring instead of from `ServletContext`. 

It could also potentially help solve the original cause of this workaround: the chicken-and-egg bean processor problem (registering bean processor in spring that needs deployment that is created from a bean that needs servlet context that needs other beans created first that spring won't allow to be created before bean processor). What do you think about that approach?

**Comment by fabiocarvalho777** [comment]
We could try third option, but not really sure yet if it would work. Also, I might not have time right now for that if it takes too long to complete. Would you be willing to try that yourself and send a pull request?

**Comment by fabiocarvalho777** [comment]
Hey Anton, after I researched a little bit more on this, I changed my mind about this situation.
As you said before, RESTEasy provides configuration via Servlet Context parameters ([as seen here(http://docs.jboss.org/resteasy/docs/3.0.16.Final/userguide/html_single/index.html#d4e127)]). So, no matter what the RESTEasy Spring Boot starter has to offer a way to have those properties configurable (since *web.xml* and *web-fragment.xml* are not possible in embedded containers).
Having said that, regardless of #7, I see this as a gap, and a also as a feature to be implemented. I am going to open a new issue for that and definitely get that done soon, hopefully for the 2.0.0 release.
